### PR TITLE
Adding 'get_same_files_recursive', 'iter_same_files_recursive'

### PR DESCRIPTION
**Pull request by BercziSandor**
- [Adding 'get_same_files_recursive', 'iter_same_files_recursive' (trivial)](https://github.com/mitar/python-deep-dircmp/commit/d424c429e774b17665a75d2a34b36cd13ec97c5d)
- [Create codeql.yml](https://github.com/mitar/python-deep-dircmp/commit/056dc97e687959b57e3ce527161f38f0d49a2b71)
- [Adding badges](https://github.com/mitar/python-deep-dircmp/commit/fa00a99cb8f5c654d73891431b17b0dd031089cd)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
 # deep-dircmp
 
+![codeql](https://github.com/BercziSandor/python-deep-dircmp/actions/workflows/codeql.yml/badge.svg)
```
Comment (mitar) [inline README.md]: This is invalid URL for this repository.

File: deep_dircmp/__init__.py [hunk 2]
```diff
@@ -65,6 +79,6 @@ def get_diff_files_recursive(self):
 
     # For our subclassed methods to be really used.
     methodmap = dict(filecmp.dircmp.methodmap,
-        subdirs=phase4,
-        same_files=phase3, diff_files=phase3, funny_files=phase3,
-    )
+                     subdirs=phase4,
+                     same_files=phase3, diff_files=phase3, funny_files=phase3,
+                     )
```
Comment (mitar) [inline deep_dircmp/__init__.py]: Please do not change style.

File: test/test_deep_dircmp.py [hunk 3]
```diff
@@ -1,3 +1,4 @@
+import os.path
```
Comment (mitar) [inline test/test_deep_dircmp.py]: Why renaming this file?

**Comment by mitar** [comment]
Otherwise it looks good. But please in the future make separate PRs for different things.

File: deep_dircmp/__init__.py (hunk 2 above)
Comment (BercziSandor) [reply deep_dircmp/__init__.py]: Auto formatted by IntelliJ, but ok.

File: README.md (hunk 1 above)
Comment (BercziSandor) [reply README.md]: Totally valid point, fixed.

File: test/test_deep_dircmp.py (hunk 3 above)
Comment (BercziSandor) [reply test/test_deep_dircmp.py]: Conventions. But ok, reverted.

File: README.md (hunk 1 above)
Comment (mitar) [reply README.md]: I do not see it yet fixed, please ping me once it is.

File: deep_dircmp/__init__.py [hunk 4]
```diff
@@ -10,7 +10,8 @@ class DeepDirCmp(filecmp.dircmp):
     """
 
     def phase3(self):
-        xx = filecmp.cmpfiles(self.left, self.right, self.common_files, shallow=False)
+        xx = filecmp.cmpfiles(self.left, self.right, self.common_files,
+                              shallow=False)
```
Comment (mitar) [inline deep_dircmp/__init__.py]: Extra change.

**Comment by mitar** [comment]
Thanks!

**Comment by mitar** [comment]
Tests are now failing in CI. Care to take a look?

**Comment by BercziSandor** [comment]
Pytest has a (strange?) file naming convention: [link](https://www.numpyninja.com/post/pytest-a-beginner-guide)

> Naming conventions of pytest
>     - Pytest file should start with either test_filename or  filename_test
>     - Naming convention for all functions should start with test_

**Pull request merged**